packages/grafana-ui/src/types/panel.ts

```diff
@@ -1,14 +1,21 @@
 import { ComponentClass } from 'react';
 import { LoadingState, SeriesData } from './data';
 import { TimeRange } from './time';
-import { ScopedVars } from './datasource';
+import { ScopedVars, DataRequestInfo, DataQueryError } from './datasource';
 
 export type InterpolateFunction = (value: string, scopedVars?: ScopedVars, format?: string | Function) => string;
 
-export interface PanelProps<T = any> {
-  data?: SeriesData[];
-  timeRange: TimeRange;
+export interface QueryResponseData {
   loading: LoadingState;
+  data: SeriesData[];
+  // TODO: annotations: ???[]; pass annotation data to panels
```

Comment:
Ah, forgotten about that. This was one of the primary reasons for the previous PanelData type (that had timeSeries and table), figured we would add annotations to that type. Trickier now that we only pass SeriesData[] to panels

Reply:
Annotations could be SeriesData - maybe we add a type property on SeriesData?

Reply:
perhaps since annotations are usually a different request and can/should be tracked independently, what about:
```
export interface QueryResponseData {
  loading: LoadingState;
  data: SeriesData[];
  
  request?: DataRequestInfo;
  error?: DataQueryError;
}

export interface PanelProps<T = any> {
  query?: QueryResponseData;
  annotation?: QueryResponseData;

  timeRange: TimeRange;
  options: T;
  renderCounter: number;
  width: number;
  height: number;
  replaceVariables: InterpolateFunction;
}
```
This is a much bigger change that would require updating all react panels.  Also `LoadingState` feels like it could be derived from (!!reqest.endTime)